jsapp/js/components/modalForms/dataAttachmentColumnsForm.es6

```diff
@@ -0,0 +1,226 @@
+import React from 'react';
+import autoBind from 'react-autobind';
+import MultiCheckbox from 'js/components/common/multiCheckbox';
+import {bem} from 'js/bem';
+import {actions} from 'js/actions';
+import {renderLoading} from '../modalForms/modalHelpers';
+
+/**
+ * Attributes from parent needed to generate `columnsToDisplay`
+ *
+ * @namespace parentAttributes
+ * @prop {string} uid
+ * @prop {string} name
+ * @prop {string} url
+ */
+
+/**
+ * The content of the DATA_ATTACHMENT_COLUMNS modal
+ *
+ * @prop {function} onSetModalTitle - for changing the modal title by this component
+ * @prop {function} onModalClose - causes the modal to close
+ * @prop {function} triggerParentLoading - triggers loading on parent modal
+ * @prop {function} generateColumnFilters - generates columns for multicheckbox
+ * @prop {object} asset - current asset
+ * @prop {parentAttributes} parent
+ * @prop {string} fileName
+ * @prop {string[]} fields - child selected fields
+ * @prop {string} attachmentUrl - if exists, we are patching an existing attachment
+                                  otherwise, this is a new import
+ */
+class dataAttachmentColumnsForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isVirgin: true,
+      isLoading: false,
+      columnsToDisplay: [],
+    };
+    autoBind(this);
+    this.unlisteners = [];
+  }
+
+  componentDidMount() {
+    // We must query for parent's asset content in order to display their
+    // available columns
+    actions.resources.loadAsset({id: this.props.parent.uid});
+
+    this.unlisteners.push(
+      actions.dataShare.attachToParent.completed.listen(
+        this.onAttachToParentCompleted
+      ),
+      actions.dataShare.attachToParent.failed.listen(
+        this.onAttachToParentFailed
+      ),
+      actions.dataShare.patchParent.completed.listen(
+        this.onPatchParentCompleted
+      ),
+      actions.dataShare.patchParent.failed.listen(
+        this.onPatchParentFailed
+      ),
+      actions.resources.loadAsset.completed.listen(
+        this.onLoadAssetContentCompleted
+      ),
+    );
+    this.setModalTitle();
+  }
+
+  componentWillUnmount() {
+    this.unlisteners.forEach((clb) => {clb();});
+  }
+
+  setModalTitle() {
+    this.props.onSetModalTitle(
+      t('Import data from ##PARENT_NAME##')
+        .replace('##PARENT_NAME##', this.props.parent.name)
+    );
```

Comment:
It's so that the strings marked for translation (with `t(...)`) make it obvious what is being handled by the code and make translators lives a bit easier, see:
https://github.com/kobotoolbox/kpi/pull/3035#discussion_r589742839